dask_cuda/tests/test_utils.py

```diff
@@ -55,9 +55,10 @@ def test_cpu_affinity():
         assert list(os.sched_getaffinity(0)) == affinity
 
 
-@pytest.mark.xfail(reason="https://github.com/rapidsai/dask-cuda/issues/313")
 def test_get_device_total_memory():
-    # Ensure Numba is using its own memory manager, rather than RMM's
+    # Close CUDA context and Ensure Numba is using its own memory manager,
+    # rather than RMM's
+    cuda.close()
     cuda.set_memory_manager(cuda.cudadrv.driver.NumbaCUDAMemoryManager)
```

Comment:
Should we call `close` again and set back to RMM at the end of the test?

```python
cuda.set_memory_manager(rmm.RMMNumbaManager)
```

Reply:
Would it be better if the behaviour of `cuda.close()` also set the memory manager back to the default? (The default being either the Numba one, or the one specified in the environment variable `NUMBA_CUDA_MEMORY_MANAGER`)

Changing that behaviour obviously wouldn't help with this PR for right now, but I wonder if this would be the behaviour that you'd hope for / expect?

Reply:
As per my previous comment, I don't think we should undo changes in a test so that some other test will be in some expected state. Instead of resetting the memory manager to RMM, another tests that require that should do it if necessary.

Reply:
> Would it be better if the behaviour of cuda.close() also set the memory manager back to the default? (The default being either the Numba one, or the one specified in the environment variable NUMBA_CUDA_MEMORY_MANAGER)

I think that's not a bad idea, what would people generally expect from such a thing? If it's expected that `cuda.close()` leaves Numba in a "clean" state, then that change would be desirable. Otherwise, maybe there could be a `cuda.reset()` or something similar to roll Numba back to its initial state?

Reply:
> I think that's not a bad idea, what would people generally expect from such a thing? If it's expected that `cuda.close()` leaves Numba in a "clean" state, then that change would be desirable. Otherwise, maybe there could be a `cuda.reset()` or something similar to roll Numba back to its initial state?

Thanks for the thoughts... I'm not quite sure what people would expect generally - `close()` is for destroying all contexts, so I would be inclined to not also reset the memory manager there. Despite what the documentation says, it turns out to be possible to change the memory manager at any time - however, existing contexts will still use their instances of memory managers, and the newly-set memory manager will only be instantiated for new contexts.

Reply:
I agree with that. I think at this time we don't have any strong argument in favor of such a change. Besides that, being explicit is probably better for sensitive functionalities like this one, so I think the way it is currently is perfectly fine.